apps/settings/js/panels/simpin_dialog/simpin_dialog.js

```diff
@@ -0,0 +1,590 @@
+/**
+ * SimPinDialog is a constructor function that can help us process
+ * any simpin related works like enable/disable lock, change pins ... etc.
+ *
+ * @module @SimPinDialog
+ */
+define(function() {
+  'use strict';
+
+  var SettingsUtils = require('modules/settings_utils');
+  var Defer = require('modules/defer');
+  var l10n = window.navigator.mozL10n;
+
+  function SimPinDialog(elements) {
+    this._localize = l10n.setAttributes;
+    this._elements = elements;
+    this._method = '';
+    this._icc = null;
+    this._cardIndex = 0;
+    this._pinOptions = {};
+    this._allowedRetryCounts = {
+      pin: 3,
+      pin2: 3,
+      puk: 10,
+      puk2: 10
+    };
+  }
+
+  SimPinDialog.prototype = {
+    /**
+     * init function
+     *
+     * @memberOf SimPinDialog
+     * @access public
+     * @param {Object} options
+     */
+    init: function(options) {
+      this._method = options.method;
+      this._cardIndex = options.cardIndex;
+      this._pinOptions = options.pinOptions;
+      this._icc = SettingsUtils.getIccByCardIndex(this._cardIndex);
+
+      if (!this._icc) {
+        console.error('We can\'t find needed icc object');
+        return;
+      }
+
+      this._bindInputClickEvent();
+      this._initUI();
+    },
+
+    /**
+     * We will call this function when users click on submit button to make sure
+     * DialogService can do the right thing.
+     *
+     * @memberOf SimPinDialog
+     * @access public
+     * @return {Promise}
+     */
+    verify: function() {
+      switch (this._method) {
+        // get PIN code
+        case 'get_pin':
+          return this._elements.pinInput.value;
+
+        // unlock SIM
+        case 'unlock_pin':
+          return this._unlockPin();
+        case 'unlock_puk':
+          return this._unlockPuk('puk');
+        case 'unlock_puk2':
+          return this._unlockPuk('puk2');
+
+        // PIN lock
+        case 'enable_lock':
+          return this._enableLock(true);
+        case 'disable_lock':
+          return this._enableLock(false);
+        case 'change_pin':
+          return this._changePin('pin');
+
+        // get PIN2 code (FDN contact list)
+        case 'get_pin2':
+          return this._updateFdnContact();
```

Comment:
based on our offline discussion, `reject` would prevent the dialog to be dismissed, so let's keep this line here.